app/src/main/java/ca/marshallasch/veil/controllers/RightMeshController.java

```diff
@@ -0,0 +1,290 @@
+
+        // Update peer list.
+        MeshManager.PeerChangedEvent event = (MeshManager.PeerChangedEvent) e;
+        if (event.state != REMOVED) {
+
+            if (event.peerUuid.equals(meshManager.getUuid())) {
+                Log.d("FOUND", "found loopback: " + event.peerUuid);
+                return;
+            }
+
+            Log.d("FOUND", "found user: " + event.peerUuid);
+
+
+
+            Sync.HashData hashData = dataStore.getDataStore();
+            Sync.MappingMessage mappingMessage =  dataStore.getDatabase();
+
+            // send messages to the peer.
+            try {
+
+                Sync.Message message = Sync.Message.newBuilder()
+                        .setType(Sync.SyncMessageType.HASH_DATA)
+                        .setData(hashData)
+                        .build();
+                meshManager.sendDataReliable(event.peerUuid, DATA_PORT, message.toByteArray());
+
+                message = Sync.Message.newBuilder()
+                        .setType(Sync.SyncMessageType.MAPPING_MESSAGE)
+                        .setMapping(mappingMessage)
+                        .build();
+                meshManager.sendDataReliable(event.peerUuid, DATA_PORT, message.toByteArray());
+            }
+            catch (RightMeshException e1) {
+                e1.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Requests data from connected peers over the mesh and sends a broadcast notifying that
+     * data has been updated.
+     */
+    public void manualRefresh(){
+        try {
+            MeshManager manager = this.meshManager;
+            Set<MeshId> peers = manager.getPeers(this.DATA_PORT);
+
+            Sync.Message dataRequest = Sync.Message.newBuilder().setType(Sync.SyncMessageType.REQUEST_DATA).build();
+
+            // request an update from everyone
+            for (MeshId peer: peers) {
+
+                // do not ask myself for info
+                if (peer.equals(manager.getUuid())) {
+                    continue;
+                }
+                manager.sendDataReliable(peer, this.DATA_PORT, dataRequest.toByteArray());
+            }
+
+        }
+        catch (RightMeshException e) {
+            e.printStackTrace();
+        }
+        Intent intent = new Intent(RightMeshController.NEW_DATA_BROADCAST);
+        LocalBroadcastManager.getInstance(serviceContext).sendBroadcast(intent);
+    }
+
+    /**
+     * Sends a set of {@link MeshId} objects as a serialized set over a local broadcast.
+     */
+    public void getPeers(){
+        Set<MeshId> peers = null;
+        try{
+            peers = meshManager.getPeers(DATA_PORT);
+        } catch (RightMeshException e){
+            e.printStackTrace();
+        }
+
+        Intent intent = new Intent("getPeers");
+        //MeshId is serializable
+        //Ref: https://developer.rightmesh.io/api/latest/reference/io/left/rightmesh/id/MeshID.php
+        intent.putExtra("peersList", (Serializable) peers);
```

Comment:
the peers list key should be a constant not a magic string

Reply:
resolved. `intent.putExtra(this.PEERS_LIST, (Serializable) peers);`